httputil/client.go

```diff
@@ -33,6 +41,58 @@ type ClientOptions struct {
 	Proxy      func(*http.Request) (*url.URL, error)
 }
 
+func addLocalSSLCertificates(conf *tls.Config) (allCAs *x509.CertPool, err error) {
+	if conf != nil && conf.RootCAs != nil {
+		allCAs = conf.RootCAs
+	} else {
+		allCAs, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("cannot read system certificates: %v", err)
+		}
+	}
+	if allCAs == nil {
+		return nil, fmt.Errorf("cannot use empty certificate pool")
+	}
+	extraCertFiles, err := filepath.Glob(filepath.Join(dirs.SnapdExtraSSLCertsDir, "*.pem"))
```

Comment:
this came to mind during the weekend,  we should probably pass this directory in at this level as an option, instead of pulling in dirs in httputil

Reply:
the struct should be setup with the dir instead of using dirs here

Reply:
I'm sure I'm missing something - the httputil/client.go file does no longer import `github.com/snapcore/snapd/dirs` it's now passed externally. But maybe you meant something different?